python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -231,6 +249,79 @@ def run(self, splash=None, version=None):
         result = app.exec_()
         return result
 
+    def uses_legacy_authentication(self):
+        """
+        Returns if the Desktop's bootrap code uses the tk-framework-login for
+        logging in.
+
+        :returns: True the bootstrap logic is older than 1.1.0, False otherwise.
+        """
+        return StrictVersion(self.app_version) < StrictVersion("1.1.0")
+
+    def create_legacy_login_instance(self):
+        """
+        Creates the pre-bootstrap-1.1.0 ShotgunLogin used to authenticate.
+
+        :raises ImportError: Thrown if the module is not available.
+
+        :returns: A tk-framework-login's ShotgunLogin instance.
+        """
+        try:
+            from python import ShotgunLogin
+        except ImportError:
+            self.log_exception("Could not import tk-framework-login")
+            raise
+        else:
+            return ShotgunLogin.get_instance_for_namespace("tk-desktop")
+
+    def _assert_authentication_fault_exists(self, splash):
```

Comment:
I think the word "restart" should be part of this method somehow. Does that make sense? Assert feels more like a check that is commented out in a release build - whereas this method is pretty drastic in comparison - it might restart the entire application! :) 

Although we are probing some stuff inside the Shotgun API, am I right thinking that it is the site config core API that is too old and this is why we are restarting? Maybe it should be called `_force_restart_on_old_core()` or something?